src/value.rs

```diff
@@ -1534,10 +1527,10 @@ impl<'a> Set<'a> {
 
 impl<'a> Function<'a> {
     /// Create a function in the current execution context for a given callback.
-    pub fn new(isolate: &'a isolate::Isolate,
+    pub fn new<'b>(isolate: &'a isolate::Isolate,
```

Comment:
Maybe it would help to have it take a `Box<Fn...>` in the signature instead?  Then you can create a box containing the function with a static lifetime.